### ISSUE TITLE
[BUG] When deleting the last characters of a word in description field it removes the space in front of the word

### ISSUE THREAD

**Issue by graue70** [TechbeeAT/jtxBoard#684]
**To Reproduce**
1. Create/open a task, go to edit mode and make sure there are some words in the description field
2. Add a single space at the end of the last word (this is only to make sure the problem appears for the last word as well)
3. Place the cursor at the end of the text (behind the space)
4. Click the delete button of the keyboard three times
5. See error

Note that the first delete action removes the space (correctly). The second delete action removes the last character of the last word (correctly). Afterwards, the remainder of the last word is underlined but the space before the word is included in the underlining.

If I type a word (without the space behind it), only the word (without the space before it) is underlined. In this case, the problem does not appear. I guess the logic of what to underline is broken when deleting, but not when typing.

**Expected behavior**
No character should be automatically removed, especially not at a position which is away from the cursor.

**Device and version:**
 - Device: Samsung Galaxy S10
 - Android version: Android 13
 - jtx Board version: 2.4.7
 - Downloaded from: f-Droid

**Comment by rogercreagh** [comment]
Yes I have the exact same problem using OpenBoard (see comment in other related bug #683). Have had it for ages.

**Comment by patrickunterwegs** [comment]
I was just reviewing the code, but the only manipulation that jtx Board does there is a check if a return was entered and the previous line starts with a bullet, then the logic would just add another bullet. 
All other fields have no logic at all when entering text. The textfields themselves are just standard Material3 textfields with no significant customization. I also can't reproduce the problem, I tried on a Samsung Galaxy S21 FE with Android 13 and an emulator with Android 13. Both just work as expected. I also tried with different keyboards (Gboard, Microsoft SwitfKey, Samsung-Keyboard), but it just always works.

To me it is very likely that this is a bug in the keyboard in combination with apps that use jetpack compose (with Material3). I assume most other apps of users are using older or different technologies and libraries, so I assume jtx Board is just a bit ahead of others. 
Even if I find a way to reproduce the issue, it is unlikely that I can find a fix for it. The code just doesn't do anything special but storing the entered value. 

So I'm sorry to say, but unless we can track it down to a specific constellation to investigate further, there's nothing that can be done here. It might also just go away with version updates of the library or of the keyboards...
